include/openrave/kinbody.h

```diff
@@ -2028,11 +2028,22 @@ class OPENRAVE_API KinBody : public InterfaceBase
         GrabbedInfo() {
         }
         bool operator==(const GrabbedInfo& other) const {
+            bool isUserDataSame = false;
```

Comment:
let's evaluate this lazily. We don't need to compare user data unless all other properties are the same, and user data might be pretty expensive to compare.

Reply:
sure. now the comparison comes the last of the other conditions.
also, I moved this `==` implementation to src/libopenrave/kinbodygrab.cpp due to the helper function header location.